// components/DeleteButton.jsx
import React from 'react'
import {object, func, string, any} from 'prop-types'

import Tooltip from '@material-ui/core/Tooltip'
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions'
import DialogTitle from '@material-ui/core/DialogTitle'
import IconButton from '@material-ui/core/IconButton'
import Button from '@material-ui/core/Button'
import DeleteIcon from '@material-ui/icons/Delete'

class DeleteButton extends React.Component {
    static propTypes = {
        title:   string.isRequired,
        className: any,
        deleteHandler: func.isRequired,
    }
    
    state = {
        open: false
    }

    render(){
        const {
            title, 
            className,
            deleteHandler,
        } = this.props
        const {open} = this.state

        return (
            <div>
                <Tooltip title={title}>
                    <IconButton 
                        onClick={() => this.setState({open: true})}
                        className={className}
                    >
                        <DeleteIcon/>
                    </IconButton>
                </Tooltip>
                <Dialog
                    open={open}
                    onClose={() => this.setState({open: false})}
                    aria-labelledby='alert-dialog-title'
                >
                    <DialogTitle id='alert-dialog-title'>{`${title}?`}</DialogTitle>
                    <DialogActions>
                        <Button onClick={deleteHandler} color='primary' autoFocus>
                            Да
                        </Button>
                        <Button onClick={() => this.setState({open: false})} color='primary'>
                            Нет
                        </Button>                        
                    </DialogActions>
                </Dialog>
            </div>
        )
    }
}

export default DeleteButton
