// components/ElementPlace.jsx
import React from 'react'
import _ from 'lodash'
import {object} from 'prop-types'

import elements from 'components/elements'

import AddIcon from '@material-ui/icons/Add'
import {withStyles} from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import grey from '@material-ui/core/colors/grey'

class ElementPlace extends React.Component {
    static propTypes = {
        element: object.isRequired,
        row:     object.isRequired,
    }
    
    state = {
        status: 'load',
        element: null,
    }

    componentWillMount = () => {
        const {element} = this.props
        if(element){
            this.setState({
                element: elements[_.findIndex(elements, el => {return el.type === element.type})]
            })
        }
    }

    render(){
        const {element} = this.state
        const {classes, row} = this.props
        if(element){            
            return (
                <Grid item xs zeroMinWidth className={classes.root}>
                    <element.edit 
                        content={this.props.element.content}
                        row={row}
                    />
                </Grid>
            )
        } else {
            return (
                <Grid item xs zeroMinWidth className={classes.root}>
                    <AddIcon className={classes.add}/>
                </Grid>
            )
        }
    }
}

const styles = theme => ({
    root: {
        //backgroundColor: theme.palette.background.paper,
        backgroundColor: grey[50],
        minHeight: 50,
        margin: '0px 20px',
        //border: `1px solid ${grey[300]}`,
        padding: '0px 10px',

    },
    add: {
        display: 'block',
        margin: '0 auto',
        marginTop: 20
    }
})

export default withStyles(styles)(ElementPlace)
