// components/Row.jsx
import React from 'react'
import {string, number} from 'prop-types'

import {withStyles} from '@material-ui/core/styles'
import grey from '@material-ui/core/colors/grey'

import Tools from 'components/Tools'
import AppendRow from 'components/AppendRow'

class Row extends React.Component {
    state = {
        status: 'load'
    }

    componentDidMount() {
        const {mode} = this.props
        this.setState({status: mode})
    }

    componentWillReceiveProps(nextProps) {
        const oldMode = this.props.mode
        const newMode = nextProps.mode
        
        if (newMode !== oldMode) {
            this.setState({status: newMode})
        }
    }

    onShowTools = val => {
        this.props.onShowTools(val)
    }
    
    render() {
        const {status} = this.state
        const {
            children, 
            classes, 
            isShowTools,
            selectRowHandler,
            id,
        } = this.props
        switch(status) {
            case 'load':
                return (
                    <div>
                        ShadowComponent
                    </div>
                )
                break
            case 'edit':
                return (
                    <div>
                        <div className={classes.root}>
                            <h3>{id}</h3>
                            {children}
                        </div>
                        <AppendRow 
                            onShowTools={this.onShowTools}
                            isShowTools={isShowTools}
                            selectRowHandler={selectRowHandler}
                            id={id}
                        />
                    </div>
                )
                break       
            case 'preview':
                return (
                    <div>                        
                        {children}
                    </div>
                )
                break
            case 'tools':
                return (
                    <div>
                        <Tools/>
                    </div>
                )
                break
        }        
    }
}

const styles = theme => ({
    root: {
        backgroundColor: grey[50],
        minHeight: 100,
        padding: 30,
    }
})

Row.propTypes = {
    mode: string.isRequired,
    id:   number.isRequired,
}

export default withStyles(styles)(Row)
