// components/Tools.jsx
import React from 'react'
import {object, func} from 'prop-types'
//import {TransitionGroup} from 'react-transition-group'
import ReactCSSTransitionGroup from 'react-addons-css-transition-group'

import {withStyles} from '@material-ui/core/styles'
import Toolbar from '@material-ui/core/Toolbar'
import grey from '@material-ui/core/colors/grey'

import rows from 'components/rows'
import elements from 'components/elements'
import ToolButton from 'components/ToolButton'
import Typography from '@material-ui/core/Typography'

class Tools extends React.Component {
    static propTypes = {
        classes:          object.isRequired,
        selectRowHandler: func.isRequired,
    }   

    render(){
        const {classes, selectRowHandler} = this.props
        return (            
            <Toolbar className={classes.root}>
                <div className={classes.panel}>
                    <ReactCSSTransitionGroup 
                        transitionName='animTranslateX'                    
                        transitionAppear={true}
                        transitionAppearTimeout={900}
                        transitionEnter={true}
                        transitionEnterTimeout={900}
                        transitionLeave={true}
                        transitionLeaveTimeout={900}                  
                    >
                        <div className={classes.title}>
                            <Typography 
                                variant='button' 
                                gutterBottom
                                color='textPrimary'
                            >
                                Выберите блок
                            </Typography>
                        </div>
                        {
                            rows.map(row => {   
                                const {type, icon} = row                                                    
                                return (                                    
                                    <ToolButton 
                                        icon={icon}
                                        key={type}
                                        type={type}
                                        clickHandler={selectRowHandler}
                                    />
                                )
                            })
                        } 
                    </ReactCSSTransitionGroup>           
                </div>
            </Toolbar>            
        )
    }
}

const styles = theme => ({
    root: {
        background: grey[600],
        border: `3px solid ${grey[500]}`,
        borderLeft: 0,
        borderRight: 0,
    },
    panel: {
        margin: '0 auto',
    },
    title: {
        color: grey[100],
        textAlign: 'center',
        marginBottom: -10,
        paddingTop: 15,
    }
})

export default withStyles(styles)(Tools)
