// components/Tools.jsx
import React from 'react'
import {object, func} from 'prop-types'
//import {TransitionGroup} from 'react-transition-group'
import ReactCSSTransitionGroup from 'react-addons-css-transition-group'

import {withStyles} from '@material-ui/core/styles'
//import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'

import rows from 'components/rows'
import elements from 'components/elements'
import ToolButton from 'components/ToolButton'

class Tools extends React.Component {
    static propTypes = {
        classes:          object.isRequired,
        selectRowHandler: func.isRequired,
    }   

    render(){
        const {classes, selectRowHandler} = this.props
        return (            
            <Toolbar className={classes.root}>
                <ReactCSSTransitionGroup 
                    transitionName='animTranslateX'                    
                    transitionAppear={true}
                    transitionAppearTimeout={900}
                    transitionEnter={true}
                    transitionEnterTimeout={900}
                    transitionLeave={true}
                    transitionLeaveTimeout={900}                  
                >
                    {
                        rows.map(row => {   
                            const type = row.type                            
                            return (                                    
                                <ToolButton 
                                    button={row.button}
                                    key={type}
                                    type={type}
                                    clickHandler={selectRowHandler}
                                />
                            )
                        })
                    } 
                </ReactCSSTransitionGroup>           
            </Toolbar>            
        )
    }
}

const styles = theme => ({
    root: {
        background: 'indianred',
    }
})

export default withStyles(styles)(Tools)
