// components/rows/Column_3.jsx
import React from 'react'

import Grid from '@material-ui/core/Grid'

import Svg from 'svg/column-3.svg'
import ElementPlace from 'components/ElementPlace'

export class Column_3 extends React.Component {    
    render(){
        const element1 = this.props.elements[0] || {}
        const element2 = this.props.elements[1] || {}
        const element3 = this.props.elements[2] || {}
        return (
            <Grid container>
                <Grid item xs={4}>
                    <ElementPlace 
                        element={element1}
                        row={this.props.row}
                    />
                </Grid>
                <Grid item xs={4}>
                    <ElementPlace 
                        element={element2}
                        row={this.props.row}
                    />
                </Grid>
                <Grid item xs={4}>
                    <ElementPlace 
                        element={element3}
                        row={this.props.row}
                    />
                </Grid>
            </Grid>
        )        
    }
}

export const preview = props => (
    <Grid container>
        <Grid container>
                <Grid item xs={4}>
                    1/3 preview
                </Grid>
                <Grid item xs={4}>
                    1/3 preview
                </Grid>
                <Grid item xs={4}>
                    1/3 preview
                </Grid>
            </Grid>
    </Grid>
)

export const button = {
    name:   'На всю ширину',    
    svg:    Svg,
}
