// components/rows/Column_2.jsx
import React from 'react'
import {array, object, func} from 'prop-types'

import Grid from '@material-ui/core/Grid'

import Svg from 'svg/column-2.svg'
import ElementPlace from 'components/ElementPlace'
import VerticalDivider from 'components/VerticalDivider'

export const Column_2 = props => {   
    const {
        elements, 
        row,
        deleteElementHandler,
        addElementHandler,
    } = props
    return (
        <Grid container>
            <Grid item xs={6}>
                <ElementPlace 
                    element={elements[0] || {}}
                    row={row}
                    rowPlace={0}
                    deleteElementHandler={deleteElementHandler}
                    addElementHandler={addElementHandler}
                />
            </Grid>
            <VerticalDivider/>
            <Grid item xs={6}>
                <ElementPlace 
                    element={elements[1] || {}}
                    row={row}
                    rowPlace={1}
                    deleteElementHandler={deleteElementHandler}
                    addElementHandler={addElementHandler}
                />
            </Grid>
        </Grid>
    )            
}

Column_2.propTypes = {
    elements:             array.isRequired,
    row:                  object.isRequired,
    deleteElementHandler: func.isRequired,
    addElementHandler:    func.isRequired,
}

export const Preview = props => (
    <Grid container>
        <Grid container>
                <Grid item xs={6}>
                    1/2 preview
                </Grid>
                <Grid item xs={6}>
                    1/2 preview
                </Grid>
            </Grid>
    </Grid>
)

export const Icon = Svg
