// components/rows/Column_3.jsx
import React from 'react'
import {array, object} from 'prop-types'

import Grid from '@material-ui/core/Grid'

import Svg from 'svg/column-3.svg'
import ElementPlace from 'components/ElementPlace'
import VerticalDivider from 'components/VerticalDivider'

export const Column_3 = props => {
    const {elements, row} = props    
    return (
        <Grid container>
            <Grid item xs={4}>
                <ElementPlace 
                    element={elements[0] || {}}
                    row={row}
                />
            </Grid>
            <VerticalDivider/>
            <Grid item xs={4}>
                <ElementPlace 
                    element={elements[1] || {}}
                    row={row}
                />
            </Grid>
            <VerticalDivider/>
            <Grid item xs={4}>
                <ElementPlace 
                    element={elements[2] || {}}
                    row={row}
                />
            </Grid>
        </Grid>
    )            
}

Column_3.propTypes = {
    elements: array.isRequired,
    row:      object.isRequired,
}

export const preview = props => (
    <Grid container>
        <Grid container>
                <Grid item xs={4}>
                    1/3 preview
                </Grid>
                <Grid item xs={4}>
                    1/3 preview
                </Grid>
                <Grid item xs={4}>
                    1/3 preview
                </Grid>
            </Grid>
    </Grid>
)

export const button = {
    name:   'На всю ширину',    
    svg:    Svg,
}
