// components/rows/Column_2.jsx
import React from 'react'
import {array, object, func} from 'prop-types'

import Grid from '@material-ui/core/Grid'

import Svg from 'svg/column-2.svg'
import ElementPlace from 'components/ElementPlace'
import VerticalDivider from 'components/VerticalDivider'

export const Column_2 = props => {   
    const {
        elements, 
        row,
        deleteElementHandler,
        addElementHandler,
    } = props
    return (
        <Grid container>
            <Grid item xs={6}>
                <ElementPlace 
                    element={elements[0]}
                    row={row}
                    rowPlace={0}
                    deleteElementHandler={deleteElementHandler}
                    addElementHandler={addElementHandler}
                />
            </Grid>
            <VerticalDivider/>
            <Grid item xs={6}>
                <ElementPlace 
                    element={elements[1]}
                    row={row}
                    rowPlace={1}
                    deleteElementHandler={deleteElementHandler}
                    addElementHandler={addElementHandler}
                />
            </Grid>
        </Grid>
    )            
}

Column_2.propTypes = {
    elements:             array.isRequired,
    row:                  object.isRequired,
    deleteElementHandler: func.isRequired,
    addElementHandler:    func.isRequired,
}

export const Preview = props => {
    const {elements} = props
    let content1 = null
    let content2 = null
    if(elements.length){
        if(elements[0] && elements[0].hasOwnProperty('content')) content1 = elements[0].content
        if(elements[1] && elements[1].hasOwnProperty('content')) content2 = elements[1].content    
    }    
    return (
        <Grid container>
            <Grid item xs={6} dangerouslySetInnerHTML={{__html: content1}}/>
            <Grid item xs={6} dangerouslySetInnerHTML={{__html: content2}}/>
        </Grid>
    )
}

export const Icon = Svg
