// components/rows/FullWidth.jsx
import React from 'react'
import {array, object, func} from 'prop-types'

import Grid from '@material-ui/core/Grid'

import Svg from 'svg/column-full.svg'
import ElementPlace from 'components/ElementPlace'

export const FullWidth = props => {
    const {
        elements, 
        row,
        deleteElementHandler,
        addElementHandler,
    } = props
    return (
        <Grid container>
            <Grid item xs={12}>
                <ElementPlace 
                    element={elements[0]}
                    row={row}
                    rowPlace={0}
                    deleteElementHandler={deleteElementHandler}
                    addElementHandler={addElementHandler}
                />
            </Grid>
        </Grid>
    )            
}

FullWidth.propTypes = {
    elements:             array.isRequired,
    row:                  object.isRequired,
    deleteElementHandler: func.isRequired,
    addElementHandler:    func.isRequired,
}

export const Preview = props => {
    const content = props.elements[0].content
    return (
        <Grid container>
            <Grid item xs={12} dangerouslySetInnerHTML={{__html: content}}/>
        </Grid>
    )
}

export const Icon = Svg
